src/kernl/optimizer/attention.py

```diff
@@ -31,8 +42,11 @@ def attention_wrapper(q, k, v, output, sm_scale, is_causal, attention_mask):
 
     # When there is a large difference between those dimensions, our kernel become inefficient
     # (almost no parallelization), so we use pytorch instead
-    if q.size(-2) == 1 and k.size(-2) > 50:
-        attention_reference(q, k, v, output, sm_scale, is_causal=is_causal, attention_mask=attention_mask)
+    if q.size(-2) == 1 and k.size(-2) > 50 and (attention_mask is None) and not is_causal:
```

Comment:
don't understand "(attention_mask is None) and not is_causal" the condition is only for attention_vec_mat_forward, for attention_reference it's ok

Reply:
you are right, I ll modify it.

Reply:
done